lib/util/promises.ts

```diff
@@ -1,27 +1,74 @@
+import is from '@sindresorhus/is';
 import pAll from 'p-all';
 import pMap from 'p-map';
+import { ExternalHostError } from '../types/errors/external-host-error';
 
 type PromiseFactory<T> = () => Promise<T>;
 
-export function all<T>(
+interface AggregateError {
```

Comment:
use `AggregateError" export from `aggregatre-error` module

![image](https://user-images.githubusercontent.com/1798109/188629963-996d7921-156e-4d8f-a4fd-afa26942e53e.png)


so we can use `AggregateError.errors` which is public and we can make `instanceof` check